pkg/scanner/utils/utils.go

```diff
@@ -4,25 +4,47 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/Masterminds/semver"
 	"github.com/aquasecurity/fanal/types"
 	"github.com/aquasecurity/trivy/pkg/log"
-	"github.com/knqyf263/go-version"
 )
 
 var (
 	replacer = strings.NewReplacer(".alpha", "-alpha", ".beta", "-beta", ".rc", "-rc")
 )
 
-func MatchVersions(currentVersion *version.Version, rangeVersions []string) bool {
-	for _, p := range rangeVersions {
-		c, err := version.NewConstraint(replacer.Replace(p))
+func MatchVersions(currentVersion *semver.Version, rangeVersions []string) bool {
+	for i := range rangeVersions {
+		rangeVersions[i] = replacer.Replace(rangeVersions[i])
+		constraintParts := strings.Split(rangeVersions[i], ",")
+		for j := range constraintParts {
+			part := strings.Split(constraintParts[j], ".")
+			if len(part) > 3 {
+				constraintParts[j] = strings.Join(part[:2], ".") + "." + strings.Join(part[2:], "-")
```

Comment:
I think it doesn't work with a patch version including a dot such as `1.2.3-beta.1.
https://github.com/Masterminds/semver#working-with-prerelease-versions

It might work even if we replace valid dots, but I think we should keep the version in its original form as much as possible to avoid an unexpected bug.

The best way I think of at the moment is 

```
1.2.3-beta -> 1.2.3-beta
1.2.3-beta.1 -> 1.2.3-beta.1
1.2.3.4 -> 1.2.3-4
1.2.3.4.5 -> 1.2.3-4.5
1.2.3.4-5 -> 1.2.3-4-5
```

Let me know your thought.

Reply:
Updated, as suggested, I think you are right, let's see if we get any other issues later